core/src/components/toast/toast.tsx

```diff
@@ -568,5 +575,9 @@ const buttonClass = (button: ToastButton): CssClassMap => {
   };
 };
 
+const buttonPart = (button: ToastButton): string => {
+  return isCancel(button.role) ? 'button-cancel' : 'button';
```

Comment:
```suggestion
  return isCancel(button.role) ? 'button cancel' : 'button';
```

For parts with state/modifiers, we should add it as multiple parts. This way the cancel button still matches the `button` part in addition to `button cancel`.